### PR TITLE
Docs for `templates apply` command

### PR DESCRIPTION
**Pull request by joshspicer**
Documenting the discussion at https://github.com/devcontainers/spec/issues/71#issuecomment-1409453050

### CODE REVIEW

**Review by samruddhikhandale — APPROVED** [review]
Awesome!

File: docs/templates/apply.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+Templates can be published via the `templates publish` command - see [the template-starter repo](https://github.com/devcontainers/template-starter) for more information.
+
+To see all the available options, run `devcontainers templates apply --help`.
+
+## Example
+
+To apply the [debian template](https://github.com/devcontainers/templates/tree/main/src/debian) to a local folder with the CLI, execute the following steps.
+
+![example](https://user-images.githubusercontent.com/23246594/215609996-c1109c72-1a05-410e-83a6-86a782cce929.png)
```
Comment (chrmarti) [inline docs/templates/apply.md:15]: I suggest to add the output as text, not as screenshot.

**Review by chrmarti — CHANGES_REQUESTED** [review]
👍 Left a request.

**Review by chrmarti — APPROVED** [review]
Great, thanks!

**Pull request merged**